src/input/plugins/CdioParanoiaInputPlugin.cxx

```diff
@@ -214,18 +214,21 @@ input_cdio_open(const char *uri,
 	}
 
 	cdio_cddap_verbose_set(drv, CDDA_MESSAGE_FORGETIT, CDDA_MESSAGE_FORGETIT);
-	if (speed > 0) {
-		FmtDebug(cdio_domain, "Attempting to set CD speed to {}x",
-			 speed);
-		cdio_cddap_speed_set(drv,speed);
-	}
 
 	if (0 != cdio_cddap_open(drv)) {
 		cdio_cddap_close_no_free_cdio(drv);
 		cdio_destroy(cdio);
 		throw std::runtime_error("Unable to open disc.");
 	}
 
+	if (speed > 0) {
```

Comment:
I don't quite get why you move this block, and your commit message doesn't explain it. Also you add a check to the `cdio_cddap_speed_set()` return value, but since you do this in the same commit that moves the code, it's hard to see.

Reply:
I've just noticed that setting the drive speed seemed to fail on my drive.
On [cd-paranoia](https://github.com/rocky/libcdio-paranoia/blob/c93c72b91585d7d06dfbd868a3ada8e69fcc164b/src/cd-paranoia.c#L1061) `speed_set` is called after `cdda_open`, so I just tried it here and it seemed to fix the issue on my side.

Then I added the check for debugging.

Reply:
Your commit message should explain this.